syntax = "proto3";

package calculatie;

// Verzoek om de prijs te berekenen voor een project
message CalculatePriceRequest {
  int32 projectId = 1; 
  int32 articleId = 2;
  string description = 3;
  string measurementType = 4;
  string measurementUnit = 5;
  float quantity = 6;
  float pricePerUnit = 7;
}

// Antwoord voor de berekening van de prijs
message CalculatePriceResponse {
  int32 projectId = 1;
  int32 articleId = 2;
  float totalPrice = 3;
}

// Verzoek om alle berekeningen voor een project op te halen
message GetProjectCalculationsRequest {
  int32 projectId = 1;
}

// Antwoord voor de berekeningen van een project
message GetProjectCalculationsResponse {
  int32 calculationId = 1;
  int32 projectId = 2;
  int32 articleId = 3;
  string description = 4;
  string measurementType = 5;
  string measurementUnit = 6;
  float quantity = 7;
  float pricePerUnit = 8;
  float totalPrice = 9;
}

// Service voor het berekenen van de prijzen
service CalculationService {
  // Bereken de prijs voor meerdere artikelen in een project
  rpc CalculateProject (stream CalculatePriceRequest) returns (stream CalculatePriceResponse);
}

// Service voor het ophalen van berekeningen voor een specifiek project
service ProjectCalculationService {
  // Haal alle berekeningen op voor een project
  rpc GetProjectCalculations (GetProjectCalculationsRequest) returns (stream GetProjectCalculationsResponse);
}
